src/Send/Memo.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import {defineMessages, useIntl} from 'react-intl'
+import {StyleSheet, Text, View} from 'react-native'
+
+import {HelperText, TextInput} from '../components'
+
+export const maxMemoLength = 256
+
+type Props = {
+  memo: string
+  onChangeText: (memo: string) => void
+}
+
+export const Memo = ({onChangeText, memo}: Props) => {
+  const strings = useStrings()
+
+  const showError = memo.length > maxMemoLength
+  const multilineMax = memo.length > 50
+
+  return (
+    <View style={styles.container}>
+      <TextInput
+        value={memo}
+        onChangeText={(memo) => onChangeText(memo.trim())}
+        label={strings.label}
+        autoComplete={false}
+        testID="memoFieldInput"
+        errorText={showError ? 'error' : undefined} // to show the error styling
+        noHelper={true}
+        style={multilineMax && styles.input}
```

Comment:
to meet the requirements

> it should be 1 row by default and then expand to max 3 rows

https://user-images.githubusercontent.com/105349292/218594223-6a989343-2e3f-4fcc-bf71-57448a360b20.mov